src/django/api/serializers.py

```diff
@@ -171,6 +172,42 @@ def get_items_url(self, facility_list):
         return reverse('facility-list-items',
                        kwargs={'pk': facility_list.pk})
 
+    def get_status(self, facility_list):
+        ERROR_STATUSES = set(
+            FacilityListItem.ERROR_STATUSES
+        )
+        PROCESSING_STATUSES = set([
+            FacilityListItem.UPLOADED,
+            FacilityListItem.PARSED,
+            FacilityListItem.GEOCODED,
+            FacilityListItem.GEOCODED_NO_RESULTS,
+        ])
+        PENDING_STATUSES = set([
+            FacilityListItem.POTENTIAL_MATCH,
+        ])
+        COMPLETE_STATUSES = set([
+            FacilityListItem.MATCHED,
+            FacilityListItem.CONFIRMED_MATCH,
+        ])
+
+        item_statuses = set(
+            facility_list.facilitylistitem_set
+            .values_list('status', flat=True)
+            .distinct())
+
+        if (item_statuses.intersection(ERROR_STATUSES)):
+            status = 'ERROR'
```

Comment:
A list can have both errors and pending items (many lists we have seen so far have some parse errors on individual lines. On average, the data is pretty dirty). We should consider returning  multiple additional "computed flags" from the serializer, one for the presence of an error and one for "waiting on you or waiting on us" processing status.

Reply:
Switched to send a distinct list of all item statuses from the back-end, and have the front-end choose messages accordingly.